admin/resources/js/components/form/Submit.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import { useFormState } from "react-hook-form";
+
+export const Submit: React.FunctionComponent<{
+  text?: string;
+  notDirtyText?: string;
+  isSubmittingText?: string;
+}> = ({ text, notDirtyText, isSubmittingText }) => {
+  const _text = text ?? "Submit";
```

Comment:
What is the purpose of adding an underscore before the variable name?

Reply:
I've seen underscores used to mark a private or internal-only version of a variable, that's what I was going for. On the other hand, they're also often used to mark unused variables, and that overloading of their use is confusing.

I've rewritten it to avoid underscores and the nested ternary for now.